packages/dds/tree/src/simple-tree/proxies.ts

```diff
@@ -583,9 +583,11 @@ function createArrayNodeProxy(
 		Object.create(arrayNodePrototype, {
 			length: {
 				get(this: TreeArrayNode) {
-					return getSequenceField(this).length;
+					fail("Proxy should intercept length");
+				},
+				set() {
+					fail("Proxy should intercept length");
 				},
-				set() {},
 				enumerable: false,
 				configurable: false,
 			},
```

Comment:
Nit: If the 'length' property on 'arrayNodePrototype' is required, I would add a comment explaining why.

If it is not required, I would remove the 'length' property here since you have good test coverage of the desired behavior to prevent regressions.

Reply:
Turns out removing this caused regressions that our test suite did not catch (I didn't trust that our test suite for length covered all the aspects of it: I had just added tests for the parts I was working on). There are a lot more ways a proxy implementation can by wrong that are obvious. (Thats part of why flex-tree was rewritten to be proxyless)

I have added more tests, focused on set and `in`, and updated and added some documentation about length, including here for why its still needed here.

I also fixed another bug with length (where set wasn't erroring).